File: frontend/src/member/domain/Member.ts
export default class Member {
  constructor(id: MemberId, name: string) {
    this.id = id;
    this.name = name;
  }

  id?: MemberId
  name: string

  static of(
    id: MemberId,
    name: string,
  ): Member {
    return new Member(
      id,
      name,
    );
  }
}

export class MemberId {
  constructor(memberId: number) {
    this.value = memberId;
  }

  value: number
}

File: backend/src/main/kotlin/org/sejin/wikihexagonal/member/domain/Member.kt
package org.sejin.wikihexagonal.member.domain

data class Member(
    val id: MemberId,
    val name: String,
) {
    companion object {
        fun of(
            id: MemberId,
            name: String,
        ) = Member(
            id = id,
            name = name,
        )
    }
}

data class MemberId(
    val value: Long,
)
